### PR TITLE
Starknet Custom Release

### PR DESCRIPTION
**Pull request by augustbleeds**
## Ticket
https://smartcontract-it.atlassian.net/browse/BCI-3447

Create the custom release for Starknet. 

Features:
* Enables nodes running Starknet to input a GasPriceSubunits pipeline and transmits gas price data on chain.
* Updates FMS and operator-ui to accept the contract account address (transmitterId is the public key)

Brings in the following cherry-picks:
https://github.com/smartcontractkit/chainlink/pull/13421
https://github.com/smartcontractkit/chainlink/pull/13536
https://github.com/smartcontractkit/chainlink/pull/13200

The following fixes have been made:
* https://github.com/smartcontractkit/chainlink/pull/13584/commits/97e8d4872e3bd941af7fe7404e8a32e904136cba (adds missing resolver that does nothing) . This is committed directly to this branch as fix

The last cherry-pick:
* As a part of https://github.com/smartcontractkit/chainlink/pull/13200, we update [chainlink-common](https://github.com/smartcontractkit/chainlink-common/commits/release/2.12.1-starknet/), [chainlink-feeds](https://github.com/smartcontractkit/chainlink-feeds/commits/release/starknet-2.12.0/), and [relayer](https://github.com/smartcontractkit/chainlink-starknet/commits/release/starknet-2.12.0/) to only pull the changes we need

So... the first 2 are cherry-picks. The 3rd is a fix. And the 4th is a cherry-pick with module updates to only pull necessary changes
image

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,8 +1,8 @@
 golang 1.21.7
 mockery 2.42.2
-nodejs 16.16.0
-pnpm 8.11.0
-postgres 13.3
+nodejs 20.13.1
+pnpm 8.15.8
+postgres 15.1
```
Comment (eutopian) [inline .tool-versions:5]: when did we move to postgres 15.1  o_O... I guess it looks like we're on 15.x in most of CI but what a big jump

Comment (augustbleeds) [reply .tool-versions:5]: i think we went from 13 -> 14 -> 15 within 2 months

**Review by eutopian — DISMISSED** [review]

File: .tool-versions (hunk 1 above)
Comment (eutopian) [reply .tool-versions:5]: nops are going to have so much fun when someone starts writing some new migration that won't run on older pg O_o... still kind of shocked and wondering what the process is even for db upgrades as it's not a trivial change depending on how much existing data is involved

Comment (augustbleeds) [reply .tool-versions:5]: .tool-versions is just used for development

Comment (augustbleeds) [reply .tool-versions:5]: but yeah, at some point if we ever use a feature that isn't on all versions we'll need our NOPs to do a database upgrade.

**Review by chris-de-leon-cll — DISMISSED** [review]

**Review by chris-de-leon-cll — APPROVED** [review]

**Review by eutopian — APPROVED** [review]

**Pull request merged**